shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java

```diff
@@ -1192,12 +1204,15 @@ public Boolean run() throws Exception {
 
   @Override
   public boolean runDistCp(List<Path> srcPaths, Path dst, Configuration conf) throws IOException {
-       DistCpOptions options = new DistCpOptions.Builder(srcPaths, dst)
-               .withSyncFolder(true)
-               .withDeleteMissing(true)
-               .preserve(FileAttribute.BLOCKSIZE)
-               .preserve(FileAttribute.XATTR)
-               .build();
+    DistCpOptions.Builder builder = new DistCpOptions.Builder(srcPaths, dst)
+            .withSyncFolder(true)
+            .withDeleteMissing(true)
+            .preserve(FileAttribute.BLOCKSIZE);
+    if (checkFileSystemXAttrSupport(dst.getFileSystem(conf))
+            && checkFileSystemXAttrSupport(srcPaths.get(0).getFileSystem(conf))) {
+      builder = builder.preserve(FileAttribute.XATTR);
+    }
+    DistCpOptions options = builder.build();
```

Comment:
What is the difference between option and params returned by constructDistCpParams? Why are we passing preserve attribute in both. Can one not suffice?